crates/sncast/src/main.rs

```diff
@@ -569,4 +585,31 @@ fn update_cast_config(config: &mut CastConfig, cli: &Cli) {
         ),
         clone_or_else!(cli.wait_timeout, config.wait_params.get_timeout()),
     );
+
+    config.url = clone_or_else!(cli.rpc_url, config.url);
+
+    let url = match cli.command {
+        Commands::Declare(declare::Declare { ref rpc_url, .. })
+        | Commands::Deploy(deploy::Deploy { ref rpc_url, .. })
+        | Commands::Call(call::Call { ref rpc_url, .. })
+        | Commands::Invoke(invoke::Invoke { ref rpc_url, .. })
+        | Commands::Multicall(Multicall {
+            command: multicall::Commands::Run(multicall::run::Run { ref rpc_url, .. }),
+        })
+        | Commands::ShowConfig(show_config::ShowConfig { ref rpc_url, .. })
+        | Commands::Script(script::Script {
+            command: script::Commands::Run(script::run::Run { ref rpc_url, .. }),
+        })
+        | Commands::TxStatus(tx_status::TxStatus { ref rpc_url, .. }) => rpc_url,
+        Commands::Account(Account { ref command }) => match command {
+            account::Commands::Add(account::add::Add { ref rpc_url, .. })
+            | account::Commands::Create(account::create::Create { ref rpc_url, .. })
```

Comment:
I think this is a sign that we should go the other way around. Instead of updating `CastConfig` and then creating a provider from it, we should create the provider in the command. If RPC is not present, try to get it from `snfoundry.toml`.